### PR TITLE
Sprint 7 qrcode

### PR DESCRIPTION
**Pull request by TZnoj**
The functionality to create a QR code for a vehicle has been added. 
By going to a vehicle's details, "Download QR code" will be available in the dropdown menu, which will the download the code. Once scanned, the code will redirect the user to the vehicle's details page

Screenshot 2022-03-10 233124
qr_code
.

### CODE REVIEW

File: frontend/src/modules/vehicles/VehicleDetails.vue [hunk 1]
```diff
@@ -307,6 +311,25 @@ export default {
           this.showMessage("Error occured while updating vehicle location", "danger");
         });
     },
+    async downloadQrCode() {
+      const url1 = "http://localhost:8080/#/inventory?vehicleSelected=" + this.vehicle._id
+      console.log(url1)
+      var src_qr = await QRCode.toDataURL(url1)
+      console.log(src_qr)
```
Comment (AbdulSir) [inline frontend/src/modules/vehicles/VehicleDetails.vue:318]: Remove Console.log

File: backend/middleware/upload.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+var storage = multer.diskStorage({
+    destination: (req, file, cb) => {
+        cb(null, 'uploads')
+    },
+    filename: (req, file, cb) => {
+        cb(null, file.fieldname + '-' + Date.now())
+    }
+});
+
+var upload = multer({storage: storage});
```
Comment (AbdulSir) [inline backend/middleware/upload.js:12]: Not sure if we need this middleware. I don't see it being referenced anywhere. Can we remove this upload.js file if we're not using it?

**Review by AbdulSir — CHANGES_REQUESTED** [review]
Code looks clean except the middleware file in the backend. Please review my comment and make sure that sonarCloud does not flag anything as a security hotspot.

File: frontend/src/modules/vehicles/VehicleDetails.vue (hunk 1 above)
Comment (TZnoj) [reply frontend/src/modules/vehicles/VehicleDetails.vue]: Removed the console.log statements

File: backend/middleware/upload.js (hunk 2 above)
Comment (TZnoj) [reply backend/middleware/upload.js]: Good point, I tested the code without the file and it doesn't impact the functionality of the project.

**Review by purplesmurf1998 — APPROVED** [review]
I've tested the functionality of the PR and everything seems to work fine, the QR code does as it is supposed to. 

Important to note that some refactoring will have to be done in the next sprint since when we deploy the application, the URL being generated in the QR code won't be http://localhost:8080

**Review by AbdulSir — APPROVED** [review]
Code looks better now. This PR is ready to be merged

**Pull request merged**